SalesforceCommerceCloud/pwa-kit
# @W-15953350 Reset Password

# Description
Revamps the Reset Password flow in the PWA Kit to leverage the SLAS password reset process and Marketing Cloud integration to send the reset password magic link to the shopper.



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- Add `usePasswordReset()` hook to the template app
- Created a demo PWA Kit Email sender for token delivery
  - `pwakitdemo@gmail.com`
  - Credentials are in 1Password in **Mobify - Product** folder

# How to Test-Drive This PR

- Navigate to Reset Password POC branch with latest changes: https://wasatch-mrt-reset-password-poc.mrt-storefront-staging.com/global/en-GB/login
  - This environment uses private client
- Initiate the reset password flow:
  - Enter your email
  - Click **Password**
  - Click **Forgot password?**
  - Enter the email associated with your account
  - ⚠️This flow occurs in 3 places, the login modal, the login page, and the checkout page. Verify reset password flow works in all 3 locations⚠️
- Check your email to verify that you received an email from `pwakitdemo@gmail.com` with a magic link, redirecting to the reset password form
- Verify that your password is reset successfully and you are redirected to the login page

To run tests:
- From `template-retail-react-app`: run `npm run test app/hooks/use-password-reset.test.js`

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
